### DISCUSSION TITLE
Use SubjectFilterModel

### DISCUSSION THREAD

**Discussion by nickboxx** [nextgenhealthcare/connect#5710] [Q&A]
I am trying to write up some code that takes in a CCD, captures some information from it, and uses the getPatientsByFilter call in order to get the MR ID from the patient model that it outputs. I am struggling to figure out however why it will not work. The only time I can get an output is when I include the setMirthResultsId line. If I don't have that, and only have a name, mrn, and oid can this work?

    var subFilterModel = new Packages.com.mirth.results.client.SubjectFilterModel;
    subFilterModel.setFirstName(firstName)
    subFilterModel.setLastName(lastName)

    var client = MirthResultsUtil.getClient(clientName);
    var result = client.getPatientsByFilter(subFilterModel)

EDIT:

I wanted to note that the whole point of this is that I would not have the Mirth Results ID yet. I am trying to take the name, oid and mrn and get the Mirth Results ID.

**Comment by jonbartels** [comment]
What is in the `server.log` for Mirth Results when you run this query?

What you're running into are undocumented but required fields or patterns. The logs and the response message from MR *should* include this data. 

If you're an MR customer you should also have platinum support with NextGen. Definitely file a case with them. If they're not helpful then consulting services from Diridium (@pacmano1 ) or Zen may be able to help more precisely and quickly than community support. 

You should close https://github.com/nextgenhealthcare/connect/discussions/5709 it's a duplicate.
